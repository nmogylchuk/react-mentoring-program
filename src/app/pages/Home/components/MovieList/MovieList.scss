@import './../../../../assets/styles/variables';

.movie-list {
  margin: 10px;

  &__count {
    margin: 5px 10px;
  }

  &__number {
    font-weight: bold;
    color: $red;
  }
}
