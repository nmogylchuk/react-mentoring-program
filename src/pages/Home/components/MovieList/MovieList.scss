@import '../../../../assets/styles/variables';

.movie-list {
  margin: 10px;

  &__count {
    margin: 20px 10px 10px 10px;
  }

  &__number {
    font-weight: bold;
    color: $red;
  }
}
