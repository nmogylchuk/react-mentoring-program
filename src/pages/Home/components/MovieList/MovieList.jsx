import React, {useState, useEffect} from 'react';
import Grid from '@material-ui/core/Grid';
import MovieItem from 'pages/Home/components/MovieItem/MovieItem';
import 'pages/Home/components/MovieList/MovieList.scss';
import movies from 'data/movies.json';

const MovieList = (props) => {
    const {handleSearchIcon} = props;
    const [moviesData, setMoviesData] = useState([]);

    useEffect(() => {
            try {
                const getMovies = setTimeout(() => {
                    setMoviesData(movies);
                }, 200);
                return () => {
                    clearTimeout(getMovies);
                }
            } catch (e) {
                console.error(e);
            }
        },
        [])

    function isEmptyArray(object) {
        return typeof object !== 'undefined' && object.length === 0;
    }

    const moviesCount = isEmptyArray(moviesData) ? 0 : moviesData.totalAmount;
    return (
        <div className='movie-list'>
            <div className='movie-list__count'><span className='movie-list__number'>{moviesCount}</span> movies
                found
            </div>
            <Grid container spacing={1} alignContent={'space-around'}>
                {isEmptyArray(moviesData) ? null : (
                    moviesData.data.map(movie => (
                        <MovieItem key={movie.id} movie={movie} handleSearchIcon={handleSearchIcon}
                                   // id={movie.id}
                                   // poster_path={movie.poster_path}
                                   // title={movie.title}
                                   // genres={movie.genres}
                                   // release_date={movie.release_date}
                                   // overview={movie.overview}
                                   // runtime={movie.runtime}
                        />

                    ))
                )}
            </Grid>
        </div>
    );
}

export default MovieList;
