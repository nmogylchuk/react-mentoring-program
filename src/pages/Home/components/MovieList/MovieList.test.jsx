import React from 'react';
import rerender from 'react-test-renderer';
import MovieList from "./MovieList";
import {movie} from '../../../../../__mocks__/mockData'

describe('when Movie List', () => {

    test('then snapshot created', () => {
        const movieListComponent = rerender.create(<MovieList movies={[movie]} handleSearchIcon={jest.fn()}/>);
        const tree = movieListComponent.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
