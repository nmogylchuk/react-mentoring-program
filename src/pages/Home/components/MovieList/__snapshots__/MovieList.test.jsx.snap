// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Movie List then snapshot created 1`] = `
<div
  className="movie-list"
>
  <div
    className="movie-list__count"
  >
    <span
      className="movie-list__number"
    />
     movies found
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-content-xs-space-around"
  >
    <div
      className="no-movie-found"
    >
      No Movie Found
    </div>
  </div>
</div>
`;
